File: Java/Unit 01/Lec 04/Q2.java
Single File Programming Question
Problem Statement



Arun has been assigned the task of writing a program that accepts user input for an Integer and a Character. The program should then use wrapper classes to convert these inputs into their corresponding primitive types (int, and char) and display the results.



Your task is to help Arun complete his assignment.

Input format :
The first line of input consists of an integer value.

The second line consists of a character.

Output format :
The first line of output prints "Integer Value: " followed by the converted primitive integer value.

The second line of output prints "Character Value: " followed by the converted primitive character value.



Refer to the sample output for formatting specifications.

Code constraints :
The given test cases fall under the following constraints:

1 ≤ integer ≤ 100000

The character will be either uppercase or lowercase.

Sample test cases :
Input 1 :
23
A
Output 1 :
Integer Value: 23
Character Value: A
Input 2 :
23
g
Output 2 :
Integer Value: 23
Character Value: g
Input 3 :
6789
P
Output 3 :
Integer Value: 6789
Character Value: P



Code



import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Integer intObj = sc.nextInt();
        Character charObj = sc.next().charAt(0);

        int intVal = intObj.intValue();
        char charVal = charObj.charValue();

        System.out.println("Integer Value: " + intVal);
        System.out.println("Character Value: " + charVal);
    }
}

File: DSA /Unit 02/Lec 09/Q2.cpp

// Problem Statement
// Prasath is developing a program to manage sorted singly linked lists of sales data. 
// The linked list should be kept sorted in ascending order. 
// The task is to insert a new data point into the sorted linked list and display the list before and after insertion.
// Help Prasath in creating the program.
// Input format :
// The first line of input contains an integer n, the number of initial data points to be inserted into the linked list. 
// The second line contains n integers, which are the initial data
// points.
// The third line contains a single integer data, the new data point to be inserted into the linked list
// Output format:
// The first line of output prints "Original data points:" 
// followed by the linked list elements before the insertion of the new data point
// The second line prints "Updated data points:" 
// followed by the linked list elements after the insertion of the new data point.
// Refer to the sample output for formatting specifications.
// Code constraints :
// The given test cases fall under the following constraints. 1<=n<=25  1<=initial data points, data <= 100
// Sample test cases :
// Input 1:
// 6
// 25 36 47 58 69 80
// 19
// Output 1:

// Original data points: 25 36 47 58 69 80
//  Updated data points: 19 25 36 47 58 69 80

// Input 2:
// 2 
// 50 100
// Output 2:
// Original data points: 50 100
// Updated data points: 50 75
//  100
// -
// 75




#include <iostream>
using namespace std;

struct Node {
    int data;
    Node* next;
};

Node* createNode(int data) {
    Node* newNode = new Node;
    newNode->data = data;
    newNode->next = NULL;
    return newNode;
}

void insertSorted(Node** head, int data) {
    Node* newNode = createNode(data);

    if (*head == NULL || (*head)->data >= data) {
        newNode->next = *head;
        *head = newNode;
        return;
    }

    Node* current = *head;
    while (current->next != NULL && current->next->data < data) {
        current = current->next;
    }

    newNode->next = current->next;
    current->next = newNode;
}

void display(Node* head) {
    Node* temp = head;
    while (temp != NULL) {
        cout << temp->data << " ";
        temp = temp->next;
    }
    cout << endl;
}

int main() {
    int n;
    cin >> n;

    Node* head = NULL;

    for (int i = 0; i < n; i++) {
        int data;
        cin >> data;
        insertSorted(&head, data);
    }

    int data;
    cin >> data;

    cout << "Original data points: ";
    display(head);

    insertSorted(&head, data);

    cout << "Updated data points: ";
    display(head);

    return 0;
}
